tests/integration/workspace_access/test_tacl.py

```diff
@@ -58,8 +59,25 @@ def test_recover_permissions_from_grants(
     # simulate: normal flow
     permission_manager = PermissionManager(sql_backend, inventory_schema, [generic_permissions, tacl])
     object_types = set()
-    for perm in permission_manager.load_all():
+    permissions_list = permission_manager.load_all()
+    for perm in permissions_list:
         object_types.add(perm.object_type)
+
     assert "TABLE" in object_types
     assert "DATABASE" in object_types
     assert "cluster-policies" in object_types
+
+    actual_raw_permissions = next(
+        obj.raw
+        for obj in permissions_list
+        if obj.object_id == dummy_table.full_name
+        and obj.object_type == "TABLE"
+        and json.loads(obj.raw)["principal"] == ws_group.display_name
+    )
+
+    assert (
+        f'{{"principal": "{ws_group.display_name}", "action_type": "MODIFY, SELECT", '
```

Comment:
Don't assert on JSON string, assert on fields. Keys will be serialized in different order each time and flake the test

Reply:
will correct

Reply:
corrected